python/plugins/processing/gui/NumberInputPanel.py

```diff
@@ -74,15 +80,64 @@ def __init__(self, number, minimum, maximum, isInteger):
             self.spnValue.setValue(0)
             self.spnValue.setClearValue(0)
 
-        self.btnCalc.clicked.connect(self.showNumberInputDialog)
+        self.btnCalc.setFixedHeight(self.spnValue.height())
+
+        self.btnCalc.clicked.connect(self.showExpressionsBuilder)
 
         self.spnValue.valueChanged.connect(lambda: self.hasChanged.emit())
 
-    def showNumberInputDialog(self):
-        dlg = NumberInputDialog(self.isInteger)
-        dlg.exec_()
-        if dlg.value is not None:
-            self.spnValue.setValue(dlg.value)
+    def showExpressionsBuilder(self):
+        context = QgsExpressionContext()
+        context.appendScope(QgsExpressionContextUtils.globalScope())
+        scope = QgsExpressionContextUtils.projectScope()
+        self.addVariablesToScope(scope)
```

Comment:
Strictly speaking you should add the projectScope intact, then add a new blank scope for the processing variables.